price-feeder/oracle/provider/bitget.go

```diff
@@ -0,0 +1,549 @@
+		return
+	}
+
+	if err := p.wsClient.WriteJSON(struct {
+		Pong uint64 `json:"pong"`
+	}{Pong: heartbeat.Ping}); err != nil {
+		p.logger.Err(err).Msg("could not send pong message back")
+	}
+}
+
+// ping to check websocket connection
+func (p *BitgetProvider) ping() error {
+	return p.wsClient.WriteMessage(websocket.PingMessage, ping)
+}
+
+func (p *BitgetProvider) setTickerPair(ticker BitgetTicker) {
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+	p.tickers[ticker.Arg.InstID] = ticker
+}
+
+func (p *BitgetProvider) setCandlePair(candle BitgetCandle) {
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+	staleTime := PastUnixTime(providerCandlePeriod)
+	candleList := []BitgetCandle{}
+	candleList = append(candleList, candle)
+
+	for _, c := range p.candles[candle.Arg.InstID] {
+		if staleTime < c.TimeStamp {
+			candleList = append(candleList, c)
+		}
+	}
+	p.candles[candle.Arg.InstID] = candleList
+}
+
+// reconnect closes the last WS connection and create a new one.
+func (p *BitgetProvider) reconnect() error {
+	err := p.wsClient.Close()
+	if err != nil {
+		return types.ErrProviderConnection.Wrapf("error closing Bitget websocket %v", err)
+	}
+
+	p.logger.Debug().Msg("reconnecting websocket")
+	wsConn, resp, err := websocket.DefaultDialer.Dial(p.wsURL.String(), nil)
+	defer resp.Body.Close()
+	if err != nil {
+		return fmt.Errorf("error reconnecting to Bitget websocket: %w", err)
+	}
+	p.wsClient = wsConn
+
+	currencyPairs := p.subscribedPairsToSlice()
+
+	telemetry.IncrCounter(
+		1,
+		"websocket",
+		"reconnect",
+		"provider",
+		string(ProviderBitget),
+	)
+	return p.subscribeChannels(currencyPairs...)
+}
+
+func (p *BitgetProvider) getTickerPrice(cp types.CurrencyPair) (types.TickerPrice, error) {
+	p.mtx.RLock()
+	defer p.mtx.RUnlock()
+
+	ticker, ok := p.tickers[cp.String()]
+	if !ok {
+		return types.TickerPrice{}, fmt.Errorf("bitget failed to get ticker price for %s", cp.String())
+	}
+
+	return ticker.toTickerPrice()
+}
+
+func (p *BitgetProvider) getCandlePrices(cp types.CurrencyPair) ([]types.CandlePrice, error) {
+	p.mtx.RLock()
+	defer p.mtx.RUnlock()
+
+	candles, ok := p.candles[cp.String()]
+	if !ok {
+		return []types.CandlePrice{}, fmt.Errorf("failed to get candles price for %s", cp.String())
+	}
+
+	candleList := []types.CandlePrice{}
+	for _, candle := range candles {
+		cp, err := candle.toCandlePrice()
+		if err != nil {
+			return []types.CandlePrice{}, err
+		}
+		candleList = append(candleList, cp)
+	}
+	return candleList, nil
+}
+
+// setSubscribedPairs sets N currency pairs to the map of subscribed pairs.
+func (p *BitgetProvider) setSubscribedPairs(cps ...types.CurrencyPair) {
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	for _, cp := range cps {
+		p.subscribedPairs[cp.String()] = cp
+	}
+}
+
+// GetAvailablePairs returns all pairs to which the provider can subscribe.
+func (p *BitgetProvider) GetAvailablePairs() (map[string]struct{}, error) {
+	resp, err := http.Get(p.endpoints.Rest + bitgetRestPath)
```

Comment:
At `GetAvailablePairs` we should hit `/api/spot/v1/public/products` get the `baseCoin` and `quoteCoin` parse that to CurrencyPair or and set the map keys with `CurrencyPair.String()` 
This way, we can know which pairs are available to be subscribed